src/main/java/reposense/util/TimeUtil.java

```diff
@@ -66,4 +66,13 @@ public static int getZoneRawOffset(ZoneId zoneId) {
         ZoneOffset zoneOffset = zoneId.getRules().getOffset(now);
         return zoneOffset.getTotalSeconds() * 1000;
     }
+
+    /**
+     * Get the {@code current} date that is in {@code fromZoneId} timezone into {@code toZoneId} timezone.
+     */
+    public static Date convertDateWithZones(Date current, ZoneId fromZoneId, ZoneId toZoneId) {
+        int zoneRawOffset = getZoneRawOffset(toZoneId);
+        int systemRawOffset = getZoneRawOffset(fromZoneId);
```

Comment:
I think the variable should not use `systemRawOffset` as it can be confused as referring to the actual system time zone.